### PR TITLE
[onert] Introduce TRIX backend

### PR DESCRIPTION
**Pull request by jyoungyun**
This commit introduces TRIX backend.
It uses trix-engine to handle trix npu device.

Signed-off-by: Jiyoung Yun <jy910.yun@samsung.com>

draft pr: #8805

### CODE REVIEW

File: runtime/onert/backend/trix/BackendContext.h [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  std::shared_ptr<DevContext> dev_context() { return _dev_context; }
+
+public:
+  // TODO Make it private
+  std::shared_ptr<TensorBuilder> tensor_builder;
+  std::shared_ptr<KernelGenerator> kernel_gen;
+
+private:
+  std::shared_ptr<DevContext> _dev_context;
```
Comment (hseok-oh) [inline runtime/onert/backend/trix/BackendContext.h:55]: Where initialize `_dev_context`?

File: packaging/nnfw.spec [hunk 2]
```diff
@@ -58,6 +59,10 @@ BuildRequires:  libjpeg-devel
 BuildRequires:  gtest-devel
 %endif
 
+%if %{trix_support} == 1
+BuildRequires:  npu-engine-devel
+%endif
```
Comment (hseok-oh) [inline packaging/nnfw.spec:64]: We need `nnfw.spec` update for both loader and backend as fast as possible. Could you make PR for this only?

Comment (jyoungyun) [reply packaging/nnfw.spec:64]: I remove this update in this PR :)

File: runtime/onert/backend/trix/Backend.h [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  std::unique_ptr<onert::backend::BackendContext> newContext(ContextData &&data) const override
+  {
+    auto &graph = *data.graph;
+    auto context = std::make_unique<BackendContext>(this, std::move(data));
+    auto tr = std::make_shared<basic::TensorRegistry>();
+    auto tb = std::make_shared<TensorBuilder>(tr);
+    context->tensor_registry = tr;
+    context->tensor_builder = tb;
+    context->kernel_gen = std::make_shared<KernelGenerator>(graph, tb, tr, context->dev_context());
```
Comment (ragmani) [inline runtime/onert/backend/trix/Backend.h:50]: A newly created `DevContext` should be passed.

Comment (jyoungyun) [reply runtime/onert/backend/trix/Backend.h:50]: The `dev_context()` function returns shared_ptr<DevContext>. So I think it is okay, am I right?

**Review by hseok-oh — APPROVED** [review]
LGTM

**Review by ragmani — APPROVED** [review]
LGTM

**Pull request merged**